### DISCUSSION TITLE
CrewAI Integration

### DISCUSSION THREAD

**Discussion by croonnicola** [langfuse/langfuse#4551] [Ideas]
### Describe the feature or potential improvement

Would be great to have a crewAI integration with Langfuse

**Comment by marcklingen** [comment]
thanks for sharing! planning to add a notebook on how to trace crewAi with langfuse soon

**Reply by tonykipkemboi** [reply]
@marcklingen, tony form crewai here. I sent a dm on X to the langfuse account about integration. we've had a good number of our customers asking for this. we already have integration with agentops and langtrace. the dm:

> ```md
> 
> The current challenge for them is the integration piece.
> 
> It looks like LangFuse only works by integrating with callback functions in agentic frameworks. 
> 
> Here's an example of LangFuse with LangChain (and it's basically the same with all other frameworks):
> ```python
> from langfuse.callback import CallbackHandler
> langfuse_handler = CallbackHandler(
>     secret_key="sk-lf-...",
>     public_key="pk-lf-...",
>     host="[https://cloud.langfuse.com](https://t.co/G8hf3KZwkW)", # 🇪🇺 EU region
>   # host="[https://us.cloud.langfuse.com](https://t.co/7q035g4SFj)", # 🇺🇸 US region
> )
>  
> # Your Langchain code
>  
> # Add Langfuse handler as callback (classic and LCEL)
> chain.invoke({"input": "<user_input>"}, config={"callbacks": [langfuse_handler]})
> ```
> 
> Right now, we (CrewAI) don't have a way to dynamically pass in additional callbacks for our CrewAgentExecutor. Most of the other observability tools track sessions like this:
> ```python
> import openlit
> 
> openlit.init(disable_metrics=True)
> ```
> or
> ```python
> from langtrace_python_sdk import langtrace
> langtrace.init(api_key='<LANGTRACE_API_KEY>')
> 
> # Now import CrewAI modules
> from crewai import Agent, Task, Crew
> ...
> ```
> Which is less intrusive and easier for users to setup.
> 
> As a path forward, are you all open to building out an integration and we can do joint marketing on this as well + add to our docs```

**Reply by marcklingen** [reply]
Thanks for reaching out and commenting here, this public channel is great!

I remember that CrewAI has some dependency on LangChain and exposed these langchain callbacks in the past, is this still a viable option?

Happy to look into the other options here as well.

**Reply by tonykipkemboi** [reply]
Thank you. So we did remove dependencies of LangChain in CrewAI.

**Comment by marcklingen** [comment]
@tonykipkemboi some big update here, there is now an opentelemetry based integration with Langfuse. It is currently in beta but we will ship many improvements to it over the coming days.

Notebook: https://langfuse.com/docs/integrations/crewai

Can we feature it in the crewai docs as well?

**Reply by jannikmaierhoefer** [reply]
@tonykipkemboi, I created a PR [here](https://github.com/crewAIInc/crewAI/pull/2100) to add this to the crewai docs.

**Comment by tonykipkemboi** [comment]
Merged and now live in our docs!

![Screenshot 2025-02-12 at 12 32 31 PM](https://github.com/user-attachments/assets/62689bfc-fbdc-4b67-ba8f-7682d44d82cf)

**Reply by marcklingen** [reply]
Thanks Tony! Ping us in case you get any feedback on this.

We’re currently planning deeper support for agents, is the CrewAI flow plot by any chance available at runtime within observability output?

crewai flow plot
